core/drivers/crypto/se050/adaptors/utils/scp_config.c

```diff
@@ -0,0 +1,370 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) Foundries Ltd. 2020 - All Rights Reserved
+ * Author: Jorge Ramirez <jorge@foundries.io>
+ *
+ * This sequence follows the Global Platform Specification 2.2 - Amendment D
+ * for Secure Channel Protocol 03
+ */
+#include <assert.h>
+#include <bitstring.h>
+#include <crypto/crypto.h>
+#include <kernel/mutex.h>
+#include <kernel/refcount.h>
+#include <kernel/thread.h>
+#include <mm/mobj.h>
+#include <optee_rpc_cmd.h>
+#include <se050.h>
+#include <se050_default_keys.h>
+#include <se050_utils.h>
+#include <scp.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include <tee_api_defines_extensions.h>
+#include <tee/tadb.h>
+#include <tee/tee_fs.h>
+#include <tee/tee_fs_rpc.h>
+#include <tee/tee_pobj.h>
+#include <tee/tee_svc_storage.h>
+#include <utee_defines.h>
+
+struct tee_scp03db_dir {
+	const struct tee_file_operations *ops;
+	struct tee_file_handle *fh;
+};
+
+static const char scp03db_obj_id[] = "scp03.db";
+static struct tee_pobj po = {
+	.obj_id_len = sizeof(scp03db_obj_id),
+	.obj_id = (void *)scp03db_obj_id,
+};
+
+static TEE_Result scp03db_delete_keys(void)__unused;
+static TEE_Result scp03db_delete_keys(void)
+{
+	struct tee_scp03db_dir *db = calloc(1, sizeof(struct tee_scp03db_dir));
+	TEE_Result res = TEE_SUCCESS;
+
+	assert(db);
```

Comment:
Should Core crash on malloc failure? Maybe return `TEE_ERROR_OUT_OF_MEMORY`?

Reply:
under want conditions should core crash? it is not clear to me...if there is no support to open TAs or perform any crypto operations should we crash? I honestly dont know...I'll go with whatever you tell me

Reply:
It's more or less the same policy as in the Linux kernel. Out of memory is almost never a valid reason.

Reply:
this is not what this assert means: this assert indicates that there is no communication possible with the cryptographic device which provides rng, rsa, huk and other operations (of course depending on config..so it might or it might not provide each one of them)

so the question you need to answer is: should the core crash if  the configured crypto device can not be accessed?  it is conceptually different from "should the core panic if a call to calloc returns null"...so vwhat is the rule being applied? I don't think there is any but I dont know.